### PR TITLE
Code review changes_jubn

### PR DESCRIPTION
**Pull request by SoftwareTesting111345**
Analysis of the program.
This is a mobile app where users can borrow books.
Was the program available in UC Github on time?
Yes
Is the program documented/commented well enough for you to understand?
Yes
Does the program compile?
Yes
Rationale behind your changes.
1-Add dependencies for koin, retrofit and things in need to build the project for future use.
2-Create AppModule.kt for the service and interface to provides an easy way for communication between the activity and the fragment or between different fragments
3-Create Interface for BookService so user can call the interface for better user experience.
Links to three specific commits that you made to your own group's GitHub repository after Sprint 1's deadline, and before Sprint 2's deadline.

https://github.com/trey368s/Mobile-Device-Programming/commit/209abc61a0126a659dbd9572976bbf59e6e2de0e
https://github.com/trey368s/Mobile-Device-Programming/commit/58a17815de68f42a786eba83df7f0d73e4d0395a
https://github.com/trey368s/Mobile-Device-Programming/commit/f2ebb69aa1ad122932e0d163a2bc608bdbc07d71
What I learned from this code review
I learnt how to create ViewModel and interface.

### CODE REVIEW

File: app/src/main/java/com/example/mylittlelibrary/service/BookService.kt [hunk 1]
```diff
@@ -10,9 +10,17 @@ import retrofit2.Response
...
 import retrofit2.create
 
-class BookService {
-    private val service = RetrofitClientInstance.retrofitInstance?.create(LibraryApi::class.java)
-    suspend fun fetchBooks(): List<Book>? {
+interface IBookService {
+    val service: LibraryApi?
+
+    suspend fun fetchBooks(): List<Book>?
+
```
Comment (SoftwareTesting111345) [inline app/src/main/java/com/example/mylittlelibrary/service/BookService.kt:19]: Create Interface for BookService. URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

File: app/src/main/java/com/example/mylittlelibrary/AppModule.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.example.mylittlelibrary
+
+import com.example.mylittlelibrary.service.BookService
+import com.example.mylittlelibrary.service.IBookService
+import org.koin.androidx.viewmodel.dsl.viewModel
```
Comment (SoftwareTesting111345) [inline app/src/main/java/com/example/mylittlelibrary/AppModule.kt:5]: Create AppModule.kt for the service and inteface. URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

File: app/build.gradle [hunk 3]
```diff
@@ -65,4 +65,26 @@ dependencies {
...
+    // Koin for Android
+    implementation "io.insert-koin:koin-android:3.1.4"
+    implementation 'com.squareup.okhttp3:okhttp:4.9.2'
+    // test
+    testImplementation 'org.jetbrains.kotlinx:kotlinx-coroutines-test:1.6.0'
+    testImplementation "io.mockk:mockk:1.12.2"
+    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
+    implementation "androidx.compose.material3:material3:1.0.0-alpha02"
+
```
Comment (SoftwareTesting111345) [inline app/build.gradle:91]: Add dependencies for koin, retrofit and things in need to build the project. URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

Comment (discospiff) [reply app/build.gradle:91]: This is quite a bit.  Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

**Comment by discospiff** [comment]
From this review:
- Koin is a good idea, if implemented fully.
- Interfaces are a good idea (SOLID).
- build.gradle should be no bigger than needed.  Only add dependencies when needed.
This branch introduces some incomplete functionality, which increases (not decreases) technical debt.  I'd hold off on merging incomplete features to main/master.

**Review by andrewglanzzz — COMMENTED** [review]
Would like someone else in the group to review this to see if this is applicable for the project!

**Pull request merged**